### ISSUE TITLE
Memory blow up when using when not flattened image array

### ISSUE THREAD

**Issue by anupamme** [tensorflow/federated#304]
Hello,

I am trying to use federated learning for my use case where I already have a densenet121 weights which were built on non-flattened image array as input.

So I tried following the mnist image classification example with some modifications for my use case. But when I do not flatten the image array I run into memory blow up.

Even with 2 training examples per client, my memory of 25GB runs out. The model weights are ~ 28 MB. The dimension of input and output are 240, 240, 3 (for image) and 6 for the label (it is a 6-dimensional vector).

So I am wondering whether there is some memory leak issue when running it without flattening of images in the input.

Thanks.

**Comment by jkr26** [comment]
Hi @anupamme,

Would you be able to provide a code snippet? Perhaps a link to a notebook on [Google Colab](https://colab.research.google.com)?

Reproducing this issue on our end should make it much easier to diagnose.

Thanks!

**Comment by anupamme** [comment]
Hello!

Follow these steps:

1. Clone https://github.com/anupamme/CheXpert-Keras
2. Install requirements.txt (some of the packages are not required for the federated code but it is easier to just install all)
3. python3.7 federated.py

Within few mins (10-15 mins) federated.py will consume all the memory. In my last execute my RAM was ~ 12G. However, I have earlier tested it on higher RAM (25G) as well.

If you are not able to run please ask.

**Comment by jkr26** [comment]
Here is a colab repro of the essentials; we'll look into the internals this week:

https://colab.research.google.com/drive/1_CxFUFinhX8jLWCreS642aophu6nRWJw

**Comment by anupamme** [comment]
@jkr26 any update on this?

**Comment by jkr26** [comment]
Hi @anupamme 

Haven't had the time to look into this yet, but we haven't forgotten.

Thanks for your interest!

**Comment by jkr26** [comment]
Finally got around to looking at this. It seems that it no longer repros; there's no problem training these models now. This may have been an issue in the dependencies that has been solved in the meantime. Closing, everything should work now.
